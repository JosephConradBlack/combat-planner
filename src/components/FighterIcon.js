import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Item } from 'semantic-ui-react';

import TextIcon from './TextIcon';

const StyledTextIcon = styled(TextIcon)`
  &&& {
    position: relative;
  }
`;

const IconWrapper = styled(Item.Extra)`
  &&& {
    display: inline-block;
    position: absolute;
    right: ${props => props.position * 33 * 2}px;
    margin-right: 5px;
  }
`;

const FighterIcon = ({ icon, number, color, position }) => {
  return (
    <IconWrapper position={position}>
      <StyledTextIcon icon={icon} number={number} color={color} />
    </IconWrapper>
  );
};

FighterIcon.propTypes = {
  icon: PropTypes.string,
  number: PropTypes.number,
  color: PropTypes.string,
  position: PropTypes.number
};

FighterIcon.defaultProps = {
  position: 0
};

export default FighterIcon;
