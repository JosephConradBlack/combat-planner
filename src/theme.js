export default {
  h1: {
    fontSize: '6em',
    fontWeight: 300,
    letterSpacing: '0.06em',
    subheader: {
      fontSize: '1em',
      fontWeight: 'lighter'
    }
  }
};
