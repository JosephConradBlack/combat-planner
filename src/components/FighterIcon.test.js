import React from 'react';
import { create } from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

import theme from '../theme';
import FighterIcon from './FighterIcon';

describe('FighterIcon', () => {
  it('renders with no props', () => {
    const component = create(
      <ThemeProvider theme={theme}>
        <FighterIcon />
      </ThemeProvider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with different icon', () => {
    const component = create(
      <ThemeProvider theme={theme}>
        <FighterIcon icon="shield" />
      </ThemeProvider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with different number', () => {
    const component = create(
      <ThemeProvider theme={theme}>
        <FighterIcon number={4} />
      </ThemeProvider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with different color', () => {
    const component = create(
      <ThemeProvider theme={theme}>
        <FighterIcon color={'blue'} />
      </ThemeProvider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with different position', () => {
    const component = create(
      <ThemeProvider theme={theme}>
        <FighterIcon position={1} />
      </ThemeProvider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
