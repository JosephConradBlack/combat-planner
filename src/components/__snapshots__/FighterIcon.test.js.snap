// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FighterIcon renders with different color 1`] = `
<div
  className="extra sc-ifAKCX gxUpRL"
  position={0}
>
  <div>
    <div
      className="extra sc-bdVaJa cdKNHG"
    >
      <i
        aria-hidden="true"
        className="blue heart outline big icon sc-bwzfXH gNTOrh"
      />
      <div
        className="sc-htpNat gmYomR"
        color="blue"
      >
        0
      </div>
    </div>
  </div>
</div>
`;

exports[`FighterIcon renders with different icon 1`] = `
<div
  className="extra sc-ifAKCX gxUpRL"
  position={0}
>
  <div>
    <div
      className="extra sc-bdVaJa cdKNHG"
    >
      <i
        aria-hidden="true"
        className="red shield big icon sc-bwzfXH gNTOrh"
      />
      <div
        className="sc-htpNat cByMRF"
        color="red"
      >
        0
      </div>
    </div>
  </div>
</div>
`;

exports[`FighterIcon renders with different number 1`] = `
<div
  className="extra sc-ifAKCX gxUpRL"
  position={0}
>
  <div>
    <div
      className="extra sc-bdVaJa cdKNHG"
    >
      <i
        aria-hidden="true"
        className="red heart outline big icon sc-bwzfXH gNTOrh"
      />
      <div
        className="sc-htpNat cByMRF"
        color="red"
      >
        4
      </div>
    </div>
  </div>
</div>
`;

exports[`FighterIcon renders with different position 1`] = `
<div
  className="extra sc-ifAKCX fCitBk"
  position={1}
>
  <div>
    <div
      className="extra sc-bdVaJa cdKNHG"
    >
      <i
        aria-hidden="true"
        className="red heart outline big icon sc-bwzfXH gNTOrh"
      />
      <div
        className="sc-htpNat cByMRF"
        color="red"
      >
        0
      </div>
    </div>
  </div>
</div>
`;

exports[`FighterIcon renders with no props 1`] = `
<div
  className="extra sc-ifAKCX gxUpRL"
  position={0}
>
  <div>
    <div
      className="extra sc-bdVaJa cdKNHG"
    >
      <i
        aria-hidden="true"
        className="red heart outline big icon sc-bwzfXH gNTOrh"
      />
      <div
        className="sc-htpNat cByMRF"
        color="red"
      >
        0
      </div>
    </div>
  </div>
</div>
`;
